### PR TITLE
[JOSS Review] Bits and Pieces - Version 2.2

### LINKED ISSUE #19: [JOSS Review] Link to FORTRAN implementation is not working
While reading the `README.md` I noticed that the link to Tsoulis' FORTRAN implementation is not working. At least I cannot access that code. Is this the right url? Or this is a problem on the SEG? I'm referring to this line: 

https://github.com/esa/polyhedral-gravity-model/blob/ebe47ecb30b946eb9a260702a096fb849611cf4d/README.md?plain=1#L55

This is not a major concern, but I decided to open this Issue it so you are aware.

---

This issue is part of the JOSS review: openjournals/joss-reviews#6384

### LINKED ISSUE #21: [JOSS Review] Minimal Python example doesn't run as expected
## The issue

While trying to run the minimal Python example showcased in the `README.md`, it runs into an error:

https://github.com/esa/polyhedral-gravity-model/blob/ebe47ecb30b946eb9a260702a096fb849611cf4d/README.md?plain=1#L76-L106 

Here I paste a copy of it that includes the fix in #20:

```python
import numpy as np
import polyhedral_gravity

# We define the cube as a polyhedron with 8 vertices and 12 triangular faces
# The density is set to 1.0
cube_vertices = np.array(
    [
        [-1, -1, -1],
        [1, -1, -1],
        [1, 1, -1],
        [-1, 1, -1],
        [-1, -1, 1],
        [1, -1, 1],
        [1, 1, 1],
        [-1, 1, 1],
    ]
)
cube_faces = np.array(
    [
        [1, 3, 2],
        [0, 3, 1],
        [0, 1, 5],
        [0, 5, 4],
        [0, 7, 3],
        [0, 4, 7],
        [1, 2, 6],
        [1, 6, 5],
        [2, 3, 6],
        [3, 7, 6],
        [4, 5, 6],
        [4, 6, 7],
    ]
)
cube_density = 1.0
computation_point = np.array([[0, 0, 0]])

potential, acceleration, tensor = polyhedral_gravity.evaluate(
    polyhedral_source=(cube_vertices, cube_faces),
    density=cube_density,
    computation_points=computation_point,
    parallel=True,
)
```

```
Traceback (most recent call last):
  File ".../example_01.py", line 37, in <module>
    potential, acceleration, tensor = polyhedral_gravity.evaluate(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 3, got 1)
```

The output of `polyhedral_gravity.evaluate()` is a list with a single element, so it cannot be unpacked into three variables.

## The solution

I would recommend to fix the example so it runs properly, unless this bug is highlighting some issue in the source code of `evaluate()` that should be tackled.

## Update 2024-03-07

I noticed that the example using `GravityEvaluable` falls in the same error as the one that uses the `evaluate()` function.

Moreover, the same examples are replicated in the `docs/quick_start_python.rst` and they won't run as expected either.

I add more points to the previous recommendation:
- Fix the `GravityEvaluable` example in `README.md` and in `docs/quick_start_python.rst`.
- Fix the `evaluate()` example in `docs/quick_start_python.rst`.
---

This issue is part of the JOSS review: openjournals/joss-reviews#6384

### LINKED ISSUE #22: [JOSS Review] Missing Ninja requirement for building from sources
## The issue

The instructions to build `polyhedral-gravity-model` from sources mention CMake as a requirement. Nevertheless, [Ninja](https://ninja-build.org/) also is required, otherwise the build process runs into an error:  

```
$ pip install .
Processing /home/santi/git/polyhedral-gravity-model
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: polyhedral_gravity
  Building wheel for polyhedral_gravity (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [56 lines of output]
      running bdist_wheel
      running build
      running build_ext
      CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!
      See also "/home/santi/git/polyhedral-gravity-model/build/temp.linux-x86_64-cpython-311/polyhedral_gravity/CMakeFiles/CMakeOutput.log".
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/santi/git/polyhedral-gravity-model/setup.py", line 148, in <module>
          setup(
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 368, in run
          self.run_command("build")
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 89, in run
          _build_ext.run(self)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 345, in run
          self.build_extensions()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 467, in build_extensions
          self._build_extensions_serial()
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 493, in _build_extensions_serial
          self.build_extension(ext)
        File "/home/santi/git/polyhedral-gravity-model/setup.py", line 138, in build_extension
          subprocess.check_call(["cmake", ext.sourcedir] + cmake_args, cwd=build_temp)
        File "/home/santi/.mambaforge/envs/polyhedral-gravity-env/lib/python3.11/subprocess.py", line 413, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '/home/santi/git/polyhedral-gravity-model', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/home/santi/git/polyhedral-gravity-model/build/lib.linux-x86_64-cpython-311/', '-DPYTHON_EXECUTABLE=/home/santi/.mambaforge/envs/polyhedral-gravity-env/bin/python3.11', '-DCMAKE_BUILD_TYPE=Release', '-DPOLYHEDRAL_GRAVITY_PARALLELIZATION=TBB', '-DLOGGING_LEVEL=2', '-DUSE_LOCAL_TBB=OFF', '-DBUILD_POLYHEDRAL_GRAVITY_DOCS=OFF', '-DBUILD_POLYHEDRAL_GRAVITY_TESTS=OFF', '-DBUILD_POLYHEDRAL_GRAVITY_LIBRARY=OFF', '-DBUILD_POLYHEDRAL_GRAVITY_EXECUTABLE=OFF', '-DBUILD_POLYHEDRAL_GRAVITY_PYTHON_INTERFACE=ON', '-DBUILD_SHARED_LIBS=OFF', '-GNinja', '--log-level=VERBOSE']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for polyhedral_gravity
  Running setup.py clean for polyhedral_gravity
Failed to build polyhedral_gravity
ERROR: Could not build wheels for polyhedral_gravity, which is required to install pyproject.toml-based projects
```

## The solution

I would recommend to add Ninja to the list of requirements needed to build the package from sources. The change should be applied to the `README.md` and the `docs/build.rst` files.

---

This issue is part of the JOSS review: https://github.com/openjournals/joss-reviews/issues/6384

### LINKED ISSUE #27: [JOSS Review] Improve tests by using densities different than 1.0
## The issue

While reading the C++ and the Python tests, I noticed that the ones that compare the results for a cube against the analytic solutions define the density of the cube to be equal to 1.0:

https://github.com/esa/polyhedral-gravity-model/blob/ebe47ecb30b946eb9a260702a096fb849611cf4d/test/python/test_polyhedral_gravity.py#L34

https://github.com/esa/polyhedral-gravity-model/blob/ebe47ecb30b946eb9a260702a096fb849611cf4d/test/calculation/GravityModelCubeTest.cpp#L29

Choosing a density equals to 1.0 that the test functions aren't checking if the density value is properly used in the code.

In the hypothetical case that a bug is introduced into this line:

https://github.com/esa/polyhedral-gravity-model/blob/ebe47ecb30b946eb9a260702a096fb849611cf4d/src/polyhedralGravity/calculation/GravityEvaluable.cpp#L65

to something like:

```cpp
        const double prefix = util::GRAVITATIONAL_CONSTANT / _density;
```

then all tests will still pass, not being able to flag this bug.

## The solution

I suggest to set the density of these cubes to a different value that could flag such bug.

---

This issue is part of the JOSS review: openjournals/joss-reviews#6384

### LINKED ISSUE #28: [JOSS Review] Sign of acceleration components?
While playing with the minimal Python example shown in the `README.md` I defined a cube in the (+,+,+) octant, so one of its vertices is in `[1, 1, 1]` and its opposite one in `[2, 2 ,2]`, with a density of 1.0. Then I defined a computation point in the origin (`[0, 0, 0]`) and used the `evaluate()` function to compute its fields.

```python
import numpy as np
import polyhedral_gravity

cube_vertices = np.array(
    [
        [1, 1, 1],
        [2, 1, 1],
        [2, 2, 1],
        [1, 2, 1],
        [1, 1, 2],
        [2, 1, 2],
        [2, 2, 2],
        [1, 2, 2],
    ]
)
cube_faces = np.array(
    [
        [1, 3, 2],
        [0, 3, 1],
        [0, 1, 5],
        [0, 5, 4],
        [0, 7, 3],
        [0, 4, 7],
        [1, 2, 6],
        [1, 6, 5],
        [2, 3, 6],
        [3, 7, 6],
        [4, 5, 6],
        [4, 6, 7],
    ]
)
cube_density = 1.0
computation_point = np.array([0, 0, 0])

potential, acceleration, tensor = polyhedral_gravity.evaluate(
    polyhedral_source=(cube_vertices, cube_faces),
    density=cube_density,
    computation_points=computation_point,
    parallel=True,
)
print(acceleration)
```

After running this code I get:
```
[-5.715199089414289e-12, -5.715199089414289e-12, -5.7151990894142846e-12]
```

The three components of the acceleration are negative, which would imply that any particle located in the computation point would feel an acceleration towards the (-,-,-) octant, meaning it would feel a repulsive force from the gravitational interaction with the cube. Given that the cube has a positive density, I'd assume that the three components should be positive, not negative.

Is this the expected result? Or are there any sign conventions that I missed?

---

This issue is part of the JOSS review: https://github.com/openjournals/joss-reviews/issues/6384

### LINKED ISSUE #29: Sign of the tensor
This is also related to issue #28
I did a small test by comparing the tensor quantities with the numerical differences of the "gravity" components (derivative of the potential) for delta = 1 m, as follows ...

d2V/dx2(x,y,z) vs. [ dV/dx(x+1,y,z) - dV/dx(x,y,z) ]
d2V/dy2(x,y,z) vs. [ dV/dy(x,y+1,z) - dV/dy(x,y,z) ]
d2V/dz2(x,y,z) vs. [ dV/dz(x,y,z+1) - dV/dz(x,y,z) ]

... and found out that the signs of the tensor do not match the delta(Vx/Vy/Vz).

On the other hand, the same also holds for the comparison with the first-order derivatives and the potential ...

dV/dz(x,y,z) vs. [ V(x,y,z+1) - V(x,y,z) ]

... the sign of e.g. Vz does not match the potential difference for z+1.

So it appears that the Vx/Vy/Vz signs are indeed flipped compared to V and the tensor. But again, as mentioned in #28, the potential of a polyhedral body should be positive for a point outside the mesh hull.

Two ideas for reasons:

A) Does the definition of the x,y,z axes matter? I used UTM (a right-handed coordinate system) both for test body and computation points with x ... Easting, y ... Northing and z ... Up? But actually I think that should not make a difference...

B) Maybe the mesh of my test body is "flipped" regarding inside/outside. But then still, the first-order derivatives do not match the potential and the tensor.

UPDATE 2024-03-11 (copied from #28):

I checked once more with the example cube from the description. My computation points were located at (0, 0, 3) and (0, 0, 3.001) above the test cube, and the signs were as follows:
V : positive (decreasing upwards) -> Vz > 0
acc[2] : positive acceleration (decreasing upwards) --> negative of Vz !
tensor[2] : positive --> correct sign of Vzz = d/dz(Vz), but flipped w.r.t. to output of acc[2]

From this I conclude that the mesh of my own test body is indeed "inverted", so that all quantities are additionally flipped.

### LINKED ISSUE #30: [JOSS Review] Documentation missing statement of need.
Just a small box to tick: for JOSS, the documentation must have:

> A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is?

I don't see such a thing in the current [documentation](https://polyhedral-gravity-model-cpp.readthedocs.io/en/stable/build.html) - just a brief paragraph about what the code does and is intended for would be fine.

This issue is part of the JOSS review: https://github.com/openjournals/joss-reviews/issues/6384

### LINKED ISSUE #32: [JOSS Review] Minor improvement in paper paragraph
## The issue

While reading the submitted paper to JOSS, I noticed that the first paragraph of the "Statement of Need" section could be improved to make it more clear:

https://github.com/esa/polyhedral-gravity-model/blob/fa2dba34709a42dec29e3fb3227c83f161693bc4/paper/paper.md?plain=1#L48

> The complex gravitational fields of irregular bodies, such as asteroids and comets, are often modeled using polyhedral gravity models since alternative approaches like mascon models or spherical harmonics struggle with these bodies' irregular geometry. The former struggles with convergence close to the surface [@vsprlak2021use], whereas the latter requires a computationally expensive amount of mascons (point masses of which the target body is composed) to model fine-granular surface geometry [@wittick2017mascon]. 

I think the usage of _former_ and _latter_ is not clear to which methods they are referring to. From the context, I understand that the _former_ should refer to the spherical harmonics approach, while the _latter_ should refer to the mascon models. If that's the case, the order of those are interchanged. Nonetheless, using _former_ here might lead the reader to think it's referring to the _polyhedral gravity models_ and not to one of the other methods.

## Possible solution

I would suggest to modify that paragraph, explicitly mentioning the methods the sentence is referring to. Something along:

```markdown
The complex gravitational fields of irregular bodies, such as asteroids and
comets, are often modeled using polyhedral gravity models since alternative
approaches like mascon models or spherical harmonics struggle with these
bodies' irregular geometry. The spherical harmonics approach struggles with
convergence close to the surface [@vsprlak2021use], whereas the mascon models
require a computationally expensive amount of mascons (point masses of which
the target body is composed) to model fine-granular surface geometry
[@wittick2017mascon].
```

I think it's ok to repeat the names of the methods in favor of clarity.

Please take this as a suggestion, and feel free to modify the paragraph at will.

---

This issue is part of the JOSS review: https://github.com/openjournals/joss-reviews/issues/6384

### LINKED ISSUE #33: [JOSS Review] Provide a list of dependencies needed to run the Jupyter Notebook
## The Issue

While trying to run the provided Jupyter Notebook locally, I noticed several dependencies need to be installed in the environment so it can actually run properly. Here I copy a non-comprehensive list of packages that are needed to run the notebook:

- `jupyter`
- `matplotlib`
- `sympy`
- `pyvista`
- `meshio`
- `tetgen`
- `openmesh`

## Possible solution

Would be nice to include a list of required dependencies in the `README.md`, or through an `environment.yml`.

---

This issue is part of the JOSS review: openjournals/joss-reviews#6384

### PR DESCRIPTION
**Pull request by schuhmaj**
# Changelog

## Documentation

 Have a look at the *new* [deployed docs as temporary site](https://schuhmaj.github.io/temporary/index.html)
(This simplifies the review process. I'll remove that once the PR is merged.)

- Resolves #19 
  - The docs & the paper contain the link to the FORTRAN implementation with a reference when it was last accessed (still hope that the website may go online in the future again)
  - In the `README.md`, I fully removed the link.
- Resolves #20 
  - Adds trailing commas
- Resolves #21 
  - Fixes `README.md` examples
  - Fixes the examples in `quick_python.rst`
- Resolves #22
  - The `ninja` dependency was hardcoded with an opt-out by setting the `CMAKE_GENERATOR` environment variable beforehand
  - This only affected the Python version when build from source with `pip`
  - Now, the build script uses the following precedences and does not fail if `ninja` is not found
    1. `CMAKE_GENERATOR` environment variable if set
    2. `ninja` if installed
    3. Any build system available on the system
- Resolves #27 
  - Introduces a new parameterized test for the cube gravity with density 42
- Resolves #30 
  - Adds the Statement of Need to the documentation (this is basically the same paragraph as the one from the JOSS paper)
  - Update the citations of Sopot 2023 since DOIs are now available
- Updates Links to Documentation (in `README.md`, `Jupyter Notebook`, `CONTRIBUTING.md`) which now resides on GitHub Pages
- Resolves #32 
- Resolves #33
  - Adds an `environment.yml` for the Jupyter Notebook
  - Also provides a PyPi-based `environment.yml` for ARM-based systems
  - State the environment-setup inside the Jupyter Notebook
- Relates to #34 
  - by providing a description of the Mesh Checking Utility in the `README.md`
- Update and Restructure the Documentation Overall!

## Implementation

- Fix Accelerations sign
  - We now follow the geodetic and geophysics convention. Hence, the potential $V$ for a polyhedron with mass $m > 0$ is positive (as it was). However, the acceleration is now inverted according to $$ \textbf{g} = + \nabla V$$
  -  Resolves #28 
  - Resolves #29

## TODOs when merged:

- [ ] Update Link to Documentation in [Conda Recipe](https://github.com/conda-forge/polyhedral-gravity-model-feedstock/blob/3f84fc08a60bd786de45095cdca6a70e7b5ebf70/recipe/meta.yaml#L58)
- [ ] Delete temporary fork with the updated docs

-----

Improvements based on https://github.com/openjournals/joss-reviews/issues/6384

### CODE REVIEW

**Review by santisoler — APPROVED** [review]
Thanks a lot for pushing all these changes @schuhmaj! I think this satisfies my requirements related to the JOSS review process.

I left one minor suggestion below, feel free to commit or edit it.

One thing I still noticed is I couldn't manage to install the Python package from sources with `pip install .` if Ninja is not installed. Even after the changes you made to `setup.py` in this branch. I don't think this is major issue for the JOSS review (I can actually install it with `ninja-build` installed in my system). But I wanted to raise the issue so you can keep investigating why this is happening. I can post some details in #22 if you desire.

Thanks again for these fixes. I enjoyed reviewing this submission.

**Review by gomezzz — COMMENTED** [review]
Hi @schuhmaj ! 

Thanks for taking care of this 🙏 However, it is extremely difficult for me with all these changes and issues mixed in one PR to discern which change is supposed to do what and to resolve which issue. 

From a quick glance, I suspect, the majority of the changes are doc changes and data changes?

If you want me to have a meaningful look I would ask you to maybe at least split it in 
- code changes
- doc changes
- For data I would also suggest to use LFS or similar. Just looking at the txts almost crashes my browser :D 

Otherwise I can only have a very cursory glance at the code as I have no idea how to connect intent of the change with the change 🙏 

Sorry for being complicated :D

**Comment by schuhmaj** [comment]
@gomezzz Sorry for the confusion; I totally agree that it got crowded. In theory, it should be sufficient to review the three "actual code changes" and review the [new docs](https://schuhmaj.github.io/temporary/index.html) (which are temporarily published under the given link to avoid the need to go through the all the `*.rst` files compared to the [current docs](https://esa.github.io/polyhedral-gravity-model/)).

(The incorporated change of santisoler is not included in the temporary build of the docs)

## Code Changes

### Actual Code Changes

- `src/polyhedralGravity/calculation/GravityEvaluable.cpp:69`  Multiply the acceleration with `-1.0` to obey the geodetic convention
- `test/calculation/GravityModelCubeTest.cpp` Add a test case for the cube with density `42.0` in addition to the existing test case using density `1.0` using the parametrized construct from GoogleTest
- `setup.py` Remove the hard dependency from `ninja` as the build system and make it only the preferred solution if it is installed


### Pseudo Code Changes

- Remove the in every header file `*.h` docstring `@param example - this is an example` the `-` in between the parameter name and explanation to improve the automated documentation by doxygen/ sphinx
- The cube reference solution `*.txt`
  - Rename the old file to `test/resources/analytic_cube_solution.txt` to `test/resources/analytic_cube_solution_density1.txt` and multiply the accelerations by `-1.0` to be coherent with the changes above
  - Add `test/resources/analytic_cube_solution_density42.txt`


## Documentation `./docs/*.rst`, README.md and CONTRIBUTING.md

This is the largest fraction of the PR. It is probably easier to review this [temporary build](https://schuhmaj.github.io/temporary/index.html).

To summarize the changes:

- State the convention (geodetic convention)
- Restructure and Optimize the Layout
- Add Statement of Need (only docs)


## Paper

- #32 
- Link to GitHub Pages instead of ReadTheDocs
- The FORTRAN reference implementation's hosting site is down. The site might come back. Instead of a complete removal of the link, I added the date of last access, #19

File: src/polyhedralGravity/calculation/GravityEvaluable.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ namespace polyhedralGravity {
 
         //10. Step: Final expressions after application of the prefix (and a division by 2 for the potential)
         potential = (potential * prefix) / 2.0;
-        acceleration = acceleration * prefix;
+        acceleration = acceleration * (-1.0 * prefix);
```
Comment (gomezzz) [inline src/polyhedralGravity/calculation/GravityEvaluable.cpp:69]: One thing I am mildly worried about is that once you push this if somebody built some code based on the old version and updates / reinstalls their env they will get wrong results without knowing about the change. It would be desirable to make their code break in that case. Maybe we could rename some function or something along the way to make this a breaking change that shows a message that the sign has changed if the they try to call the old version on this version?

**Review by gomezzz — APPROVED** [review]
Thanks for elaborating! Okay, that actually makes it pretty tangible. My only remaining concern now is re the comment I left of introducing a non-breaking change that still fundamentally changes results. We can address it here or maybe it is better to merge this and have a separate issue + PR for it?

File: src/polyhedralGravity/calculation/GravityEvaluable.cpp (hunk 1 above)
Comment (schuhmaj) [reply src/polyhedralGravity/calculation/GravityEvaluable.cpp:69]: Overall, I agreed; however, I am unsure about how.
A `warning` or `rename/ depcreation` seems un-intuitive.

The best I could imagine would be something like a `verbose` option being `True` by default. The option could provide information about recent changes but also enable logging.
`evaluate(..., verbose=False)`.
So, basically, a new feature.

On the other hand, simply having more options also seems a bit counterintuitive and contradicts the "slim, easy-to-use interface idea."

Not sure. Anyway, I will merge this so that we can get the JOSS paper rolling.

The `verbose` option might be something that fits nicely with #34, which will adapt the signature.
For now, I hope that a user who notices a different behavior's first idea is to check the Changelog—and there, I will definitely put the change in bold letters (Also, the package description now states which convention we follow).

**Comment by schuhmaj** [comment]
Thanks a lot for the review **santisoler** and **gomezzz** 🙂

Let's get this merged!
